be/src/olap/short_key_index.h

```diff
@@ -0,0 +1,183 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <vector>
+
+#include "common/status.h"
+#include "gen_cpp/column_data_file.pb.h"
+#include "gen_cpp/olap_file.pb.h"
+#include "olap/field_info.h"
+#include "olap/file_helper.h"
+#include "olap/row_cursor.h"
+#include "olap/olap_index.h"
+#include "util/faststring.h"
+#include "util/slice.h"
+
+namespace doris {
+
+// Used to encode a segment short key indices to 
```

Comment:
> incomplete sentence. Besides, it would be helpful to provide general information about the motivation and design (why and what) of short key index at the beginning of the file. Builder, decoder, and iterator of short key indexes are just components that implements short key index.

OK, I will finish the comment.